### PR TITLE
Eliminate bias toward GCR as Docker registry

### LINKED ISSUE #77: Proposal: Eliminate GCR bias
This replaces #67 

The `docker` (Docker build) and `push` (Docker push) `make` targets currently require either a GCP project ID or GCR repository URL to be provided. The `docker` target uses this information when tagging images that are subsequently pushed by the `push` target, however, I propose this bias toward GCR is a hindrance for anyone wishing to use another Docker registry-- Docker Hub, quay.io, a local registry, or other.

Why this is important: The beauty of Docker and k8s is that they run on most any platform, yet give you (more or less) a uniform experience across those many platforms. Expressing a bias toward one vendor in particular harms the developer experience by raising the barrier for entry vis a vis project contribution by requiring familiarity with and a budget for the vendor in question. i.e. What if I have no money and want to push images to a local registry that my minikube can pull from?

What I'd like to propose is we adopt something similar (not necessarily identical) to how we have long managed this issue at Deis. At Deis, all our build processes respect an environment variable `DEIS_REGISTRY`. When building and tagging images, the image's unqualified name (a component name like "router" or "controller") is prefixed with the value of the `DEIS_REGISTRY` environment variable so that a subsequent push uploads the image to the user-specified location.

I would further propose that we maximize the simplicity of this equation by _not_ building in any specific support for authenticating to registries. I propose instead that we simply document the requirement that anyone using the `push` target must already be authenticated to whatever registry it is they intend to push to.

### PR DESCRIPTION
**Pull request by krancour**
Closes #77 

After these changes, I am able to (easily) push images to a local registry and deploy to minikube-- although there's nothing minikube-specific about this.

### CODE REVIEW

**Comment by martinmaly** [comment]
Thanks! LGTM

**Comment by arschles** [comment]
LGTM

File: script/Makefile.mk [hunk 1]
```diff
@@ -42,26 +41,20 @@ else
...
 #
-ifneq ($(origin PROJECT),undefined)
-GCR ?= gcr.io/$(PROJECT)/catalog
-endif
-
-#
-# If GCR is defined, prepare to push images.
-#
-ifeq ($(origin GCR),undefined)
+ifeq ($(origin REGISTRY),undefined)
```
Comment (duglin) [inline script/Makefile.mk]: can we not just let it default to dockerhub?

Comment (duglin) [inline script/Makefile.mk]: or I should say... let it default to whatever "docker push" defaults to.

Comment (arschles) [inline script/Makefile.mk]: `docker push` defaults to dockerhub iff you don't specify a registry. afaik that's the only way to use that registry (e.g. you can't explicitly specify that registry). The only way to override that default is to explicitly specify the registry you want to use. Hopefully that wasn't too convoluted and makes some sense.

Comment (martinmaly) [inline script/Makefile.mk]: I like that behavior. I would worry if unqualified 'make push' actually tried to push somewhere as that could cause unintended overwriting of things. What @krancour proposes seems like perfect thing - it is explicit but very usable, and unbiased.

**Comment by krancour** [comment]
Rebasing...

File: script/Makefile.mk (hunk 1 above)
Comment (duglin) [inline script/Makefile.mk]: yep - all good!

**Comment by duglin** [comment]
LGTM

**Pull request merged**